ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/party_management_service.proto

```diff
@@ -112,6 +112,12 @@ message AllocatePartyRequest {
   // have to be unique.
   // Optional
   string display_name = 2;
+
+  // If set to true, then the party will be allocated in read-only mode.
+  // Read-only parties may not authorize commands. They are useful for allowing other
+  // parties to read contract data on their behalf (see ``read_as`` in ``commands.proto``).
+  // Optional
+  bool read_only = 3;
```

Comment:
@rautenrieth-da : two questions
1. does this mean that read-only parties can only ever be observers?
2. for something like a `public` party, we'd need more than this wouldn't we? We'd need read-only parties to be allocated on multiple nodes?

Reply:
1. The intention is that read-only parties can be used as `read_as` parties, but not as `act_as` parties in command submissions. As `read_as` parties are only used for visibility checks, those parties will not be used for anything in the transaction submitted to the ledger. The intention is also that any read-only party is hosted in read-write mode on another participant, which will submit transactions where the party is at least observer.
2. Yes, on a distributed ledger, a "public" party would need to be allocated on all nodes. Note that the corresponding epic (Multi-node party allocations) was deprioritized recently. On a single-node ledger, a "public" party would need to be allocated in read-write mode. 

So in short, read-only parties without multi-node party allocations don't add any value.